src/sonic-config-engine/minigraph.py

```diff
@@ -157,11 +161,94 @@ def parse_png(png, hname):
+                # information in ASIC ports table
+                if chassis_internal.lower() == "false":
+                    # if chassis internal is false, the interface name will be
+                    # interface alias which should be converted to asic port name
+                    if (enddevice.lower() == hostname.lower()):
+                        if ((port_alias_asic_map.has_key(endport)) and
+                                (asic_name.lower() in port_alias_asic_map[endport].lower())):
+                            endport = port_alias_asic_map[endport]
+                            neighbors[port_alias_map[endport]] = {'name': startdevice, 'port': startport}
+                            device_neighbors.append(startdevice)
+                            if bandwidth:
+                                port_speeds[port_alias_map[endport]] = bandwidth
+
+                    if (startdevice.lower() == hostname.lower()):
+                        if ((port_alias_asic_map.has_key(startport)) and
+                               (asic_name.lower() in port_alias_asic_map[startport].lower())):
+                            startport = port_alias_asic_map[startport]
+                            neighbors[port_alias_map[startport]] = {'name': enddevice, 'port': endport}
+                            device_neighbors.append(enddevice)
+                            if bandwidth:
+                                port_speeds[port_alias_map[startport]] = bandwidth
+                else:
+                    if ((enddevice.lower() == asic_name.lower()) and
+                        (startdevice.lower() != hostname.lower())):
+                        if port_alias_map.has_key(endport):
+                            endport = port_alias_map[endport]
+                        neighbors[endport] = {'name': startdevice, 'port': startport}
+                        device_neighbors.append(startdevice)
+                        if bandwidth:
+                            port_speeds[endport] = bandwidth
+
+                    if ((startdevice.lower() == asic_name.lower()) and
+                        (enddevice.lower() != hostname.lower())):
+                        if port_alias_map.has_key(startport):
+                            startport = port_alias_map[startport]
+                        neighbors[startport] = {'name': enddevice, 'port': endport}
+                        device_neighbors.append(enddevice)
+                        if bandwidth:
+                            port_speeds[startport] = bandwidth
+
+        if child.tag == str(QName(ns, "Devices")):
+            for device in child.findall(str(QName(ns, "Device"))):
+                (lo_prefix, mgmt_prefix, name, hwsku, d_type, deployment_id) = parse_device(device)
+                device_data = {'lo_addr': lo_prefix, 'type': d_type, 'mgmt_addr': mgmt_prefix, 'hwsku': hwsku }
+                if deployment_id:
+                    device_data['deployment_id'] = deployment_id
```

Comment:
why do we need deployment_id here?

Reply:
Removed deployment_id from parse_asic_meta

Reply:
the front end interface neighbors can have deployment ID.